statsmodels/gam/smooth_basis.py

```diff
@@ -839,10 +842,13 @@ class BSplines(AdditiveGamSmoother):
         underlying explanatory variable for smooth terms.
         If 2-dimensional, then observations should be in rows and
         explanatory variables in columns.
-    df :  int
-        numer of basis functions or degrees of freedom
-    degree : int
-        degree of the spline
+    df :  {list[int], int}
```

Comment:
{int, array_like[int]} is more precise since np.array(...,dtype=int) is a good input.